samples/api/hello_triangle/hello_triangle.cpp

```diff
@@ -207,11 +207,33 @@ void HelloTriangle::init_instance(Context &                        context,
 	std::vector<const char *> active_instance_layers(required_instance_layers);
 
 #ifdef VKB_VALIDATION_LAYERS
-	active_instance_layers.push_back("VK_LAYER_GOOGLE_threading");
-	active_instance_layers.push_back("VK_LAYER_LUNARG_parameter_validation");
-	active_instance_layers.push_back("VK_LAYER_LUNARG_object_tracker");
-	active_instance_layers.push_back("VK_LAYER_LUNARG_core_validation");
-	active_instance_layers.push_back("VK_LAYER_GOOGLE_unique_objects");
+	// Optimal validation layers
```

Comment:
I can foresee edits being made to this in the future and for debugging purposes. Since we've written this twice, could we potentially move it to a `vk_common.*` function that handles this entire process for us?

Reply:
I can, but the point of hello_triangle.cpp appears to be avoiding delegating functionality like that to external files.  Most of what's in that file is duplicated logic.  If you still want me to put the logic in a common place, let me know and I'll do that.

Reply:
Yes it is, it's meant to be a one file solution, and not have much code outside, however in practice this is much harder to achieve. 

I would argue selection logic for validation layers is something that wouldn't be necessary to keep in the file. 

If we had a function written that was like: `std::vector<const char*> validation_layers = select_validation_layers(available_validation_layers, requested_validation_layers)` then I think this is sufficient, and if the developer wanted to step into the code they are very much welcome too.

Reply:
This, or we keep it in the file but simplify the selection logic for `hello_triangle` (separate to `instance.*`) to a list of known layers and we just select any that are available